\chapter{Discussion and summary}
\chaptermark{Discussion and summary}

The work I have done for this thesis does not, by any means, close the book on our understanding of selection in the house mouse. Indeed, my analyses point towards a number of different possible avenues for further research. In the following section I will describe several 

I have assumed that all CNEs share a single DFE which is a first approximation, but is unrealistic. The method used by Halligan et al (2013) to identify the CNEs I analysed throughout my thesis successfully identifies known regulatory features in the genome. There are multiple roles played by the CNEs identified using phastCons, or similar approaches. These include the regulation of gene expression and the control of alternate splicing. It seems reasonable to expect that these two classes of elements, which have very different biochemical roles, will be subject to different DFEs. Dividing up CNEs that have been identified using phylogenetic methods into different functional categories represents a challenge. Data of the kind generated by ENCODE could be used to assign biochemical function to CNEs identified using phylogenetic approaches, i.e. one could take the intersection of ENCODE data and phastCons elements.

\section{Incorporating the effects of selection at linked sites when analysing the uSFS}

In chapters 2 and 3 I made use of two methods to estimate the distribution of fitness effects by analysis of the uSFS, DFE-alpha and polyDFE. Both DFE-alpha and polyDFE make use of a class of putatively neutral, linked sites to account for distortions in the shape of the uSFS away from  neutral expectation caused by processes other than the direct effects of selection. In the case of DFE-alpha, an explicit demographic model is fit to the neutral uSFS. Processes other than population size change may distort the uSFS, in particular selection at linked sites. To account for this I applied a demographic correction when using DFE-alpha (polyDFE, on the other hand, uses the neutral sites' uSFS to estiamte a set of  nuisance parameters). However, both the demographic correction applied in DFE-alpha analyses and the nuisance parameters in polyDFE essentially throw away information. The processes of background selection and selective sweeps distort the uSFS in ways that are dependant on the DFE (REF DUMP). 
Incorporating this into methods to estimate the DFE would make use of more of the information present in the genome. The uSFS analysis methods developed by Tataru et al, Glemin, Barton and Zeng model variation in the mutation rate. This is achieved by separating the genome into chunks that 
A potential avenue for investigation is incorporating these approximations into analysis of the uSFS. Information from a class of neutral sites, distant to functional elements could be used to to either parametrise a demographic model, or obatin nuisance parameters. A 



\section{Adaptation in regulatory and protein-coding portions of the mouse genome}



\section{Moving beyond mice}

Mice are an excellent model organism for studies of mammalian molecular evolution. Obviously, the genomic resources available for mice (reference genome, annotations  kit available for studies in mice is close to unrivalled. 
However, it remains to be seen whether the conclusions that we reached in this thesis can be generalised to other organisms. 
 
Throughout this thesis, we have examined evidence for molecular evolution at several levels; from the very broad (e.g. looking at the correlation between recombination rate and genetic diversity in Chapter 2) to the very precise (analysis of the uSFS in Chapter 3). In this thesis we have come to a number of conclusions. 

In other 

