\chapter{Introduction}
\chaptermark{Introduction}

\emph{Portions of this introduction have been published as a review article in BMC Biology: \\CITATION\\My contributions to that review have been reproduced here with slight modifications to the text.}

\section{Using models of selective sweeps to estimate positive selection parameters}
 
If adaptive substitutions are common, selection is expected to leave footprints in genetic diversity at linked sites. In particular, as a positively selected mutation increases in frequency, it tends to reduce diversity at linked neutral loci. Theoretical analysis of this process, termed a selective sweep (Box 1), has shown that the reduction in diversity at a linked neutral locus depends on the ratio of the strength of positive selection to the recombination rate. Thus, comparing diversity at multiple neutral loci linked to selected regions, in principle, should provide an indirect means for estimating parameters of positive selection.
 

If a population experiences recurrent selective sweeps, there are several patterns predicted by theory. Under recurrent hard selective sweeps, levels of genetic diversity are expected to be lower i) in regions of the genome with restricted recombination, ii) in regions experiencing many sweeps and iii) in the genomic regions surrounding the targets of selection themselves. Each of these of these predictions have been met in empirical studies, and each has been used to estimate parameters of positive selection.


\subsection{The Correlation Between Diversity and the Rate of Recombination}

In the late 1980s, evidence began to emerge suggesting that genetic polymorphism are less frequent in genomic regions experiencing restricted crossing-over (Aguade et al. 1989; Stephan and Langley 1989). Soon after, Begun and Aquadro (1992) showed that there is a positive correlation between nucleotide diversity and the rate of crossing-over in \emph{D. melanogaster}, a pattern subsequently observed in other eukaryotic species (Cutter and Payseur 2013). Begun and Aquadro pointed out that the correlation is qualitatively consistent with the action of recurrent selective sweeps. Wiehe and Stephan (1993) formulated expressions, based on the correlation between nucleotide diversity and the rate of recombination, to estimate the compound parameter% $\lambda2N_{e}s$, where $\lambda$ is the rate of sweeps per base pair per generation, Ne is the effective population size and s is the selection coefficient. They applied their method to the data of Begun and Aquadro (1992), estimating $\lambda2N_{e}s$ = 5.37 x 10^{-8}, but their method could not disentangle the individual parameters. More recently, Coop and Ralph (2012) performed a similar analysis in D. melanogaster to explore the effects of partial sweeps on parameter estimates. They showed that when partial sweeps are common, the rate of adaptive evolution is underestimated if the hard sweep model is assumed.
 
The correlation between diversity recombination observed by Begun and Aquadro (1992) can also be explained by background selection, the reduction in neutral diversity caused by the removal of linked deleterious mutations (Charlesworth et al. 1993). The process of background selection is qualitatively similar to recurrent selective sweeps, since both processes reduce local genetic diversity (Charlesworth 2009) and skew the SFS towards rare variants (Braverman et al. 1995; Charlesworth et al. 1995). Models of background selection envisage a neutral site linked to many functional sites at different distances, such that the effects of selection accumulate to reduce diversity (Hudson and Kaplan 1995; Nordborg et al. 1996). The correlation between neutral diversity and the recombination rate predicted by background selection is quantitatively similar to that observed in D. melanogaster (Charlesworth 1996). Indeed, recent studies suggest that background selection is a major determinant of nucleotide diversity variation at broad scales (>100Kbp) in humans (McVicker et al. 2009) and D. melanogaster (Charlesworth 2012; Comeron 2014). It is clear, then, that background selection is a key confounding factor when attempting to make inferences about positive selection.
 
\subsection{Correlation Between Neutral Diversity and Non-Neutral Divergence}

If there is a constant fraction of adaptive substitutions, α, across the genome for a given class of sites, regions that evolve at higher rates should experience a greater number of selective sweeps. Under a model of recurrent sweeps, it follows that there should be a negative correlation between nucleotide divergence at selected sites and diversity at linked neutral sites. This was first described in Drosophila melanogaster by Andolfatto (2007), and has been subsequently reported in other Drosophila species (Haddrill et al. 2011). Assuming a single rate of sweeps (λ) and a constant scaled strength of positive selection (2Nes) for a given class of sites, Andolfatto (2007) generalised formulae of Wiehe and Stephan (1993) based on the correlation between synonymous site diversity and non-synonymous site divergence to estimate  λ2Nes = 3 x 10-8 for the X-chromosome in D. melanogaster. Note that this λ2Nes estimate is similar to that obtained based on the correlation of synonymous site diversity and recombination rate (Wiehe and Stephan 1993; see above). Using an estimate of α = 0.50 obtained from a MK-based analysis, Andolfatto (2007) decomposed the λ2Nes compound parameter, and inferred that s ~ 0.001% and λ = 3.6 x 10-11 /bp/generation, suggesting that adaptation of protein-coding genes in D. melanogaster is driven by moderately weak selection (i.e., assuming D. melanogaster Ne =106, 2Nes ~ 40). In a related study, Macpherson et al. (2007) estimated λ2Nes ~ 10-7 in D. simulans, also by examining the correlation between mean neutral diversity and selected (nonsynonymous) divergence. However, their model also included the heterogeneity in levels of diversity, which is related to the rate and strength of sweeps in a different way to the mean, and allowed the individual parameters to be fitted by regression. The estimates of the compound parameter λ2Nes are similar between the two studies, though Macpherson et al. (2007) estimated that s ~ 1% (compared to Andolfatto’s estimate of s ~ 0.001 %) and λ = 3.6 x 10-12 /bp/generation. The discrepancies between the studies may be due to differences in biology between the species, or may reflect methodological differences: For example, if the majority of adaptive substitutions are driven by weakly selected sweeps, which will leave a relatively small signal in levels of  polymorphism, the MK-based method may more sensitively detect them, perhaps explaining the higher rate of sweeps inferred by Andolfatto (2007). On the other hand, strongly selected sweeps will leave a larger footprint in levels of diversity, so will be more readily detected using the approach of Macpherson et al. (2007), perhaps explaining why they inferred a lower overall rate of sweeps, with higher selection coefficients (for a full description, see Sella et al. 2009). In both cases, inferences based on variation in polymorphism may reflect processes other than the fixation of adaptive alleles that have gone to fixation, such as partial sweeps and background selection, as these will affect patterns of diversity but not necessarily divergence. Related to this, the approach employed by Andolfatto (2007) has recently been extended by Campos et al. (submitted), by estimating the correlation between synonymous site diversity and non-synonymous divergence in the presence of both background selection and gene conversion in D. melanogaster. They found that ignoring background selection tends to increase and decrease estimates of selection strength and rate, respectively. The parameter values estimated in their study suggest that 0.02% of new mutations at nonsynonymous sites are strongly selected (s ~ 0.03%, assuming Ne = 106 for D. melanogaster).
 
\subsection{ Patterns of Diversity Around the Targets of Selection}
 
An individual hard selective sweep is expected to leave a trough in genetic diversity around the selected site. If a large proportion of amino acid substitutions are adaptive, as suggested by MK-type analyses (see above), collating patterns of diversity around all substitutions of a given type should reveal a trough in diversity. Such a pattern is not expected around a “control” class of sites, such as synonymous sites. This test, proposed by Sattath et al. (2011), was first applied it to D. simulans, and the above pattern was found. By fitting a hard sweeps model to the shape of the diversity trough, they estimated α values of  5\% and 13\%, depending on whether one or two classes of beneficial mutational effects were fitted. Note that their estimates of α are substantially lower than those obtained using MK-based methods for D. melanogaster (Andolfatto 2007). Sattath et al. (2012) suggested that modes of selection other than hard sweeps may help explain to this discrepancy. However, even when modelling two classes of beneficial mutations, they found that amino acid substitutions are driven by strongly adaptive mutations (s $sim0.5\%$ and s $\sim0.01\%$). Their estimates of selection strength are therefore in broad agreement with the estimate of s $sim1\%$ obtained by Macpherson et al. (2007), based on the correlation between synonymous diversity and non-synonymous divergence in D. simulans. The Sattath et al. (2012) test, then, suggests that adaptation in protein-coding genes is fairly frequent and driven by strong, hard sweeps.
 
The Sattath test has been applied in a variety of organisms, including humans (Hernandez et al. 2011), wild mice (Halligan et al. 2013), Capsella grandiflora (Williamson et al. 2014) and maize (Beissinger et al. 2016). In all but C. grandiflora, researchers have found no difference in patterns of diversity around selected and neutral substitutions. These results have been interpreted as evidence that hard sweeps were rare in the recent history of both humans (Hernandez et al. 2011) and maize (Beissinger et al. 2016). However, Enard et al. (2014) pointed out that the Sattath test will be underpowered if there is large variation in levels of functional constraint in the genome. Indeed, through their analyses Enard et al. (2014) found evidence for frequent adaptive substitutions in humans, particularly in regulatory sequence. To address the issues raised by Enard et al. (2014), Beissinger et al. (2016) applied the Sattath test to substitutions in maize genes with the highest and lowest levels of functional constraint separately, but still found no difference in diversity pattern, suggesting either that hard sweeps have been rare in that species or that there is another confounding factor.
 
One possible explanation is that the species in which the Sattath test did/did not detect hard sweeps have distinct patterns of linkage disequilibrium (LD). LD decays to background levels within hundreds of base-pairs in D. simulans (Langley et al. 2012) and C. grandiflora (Josephs et al. 2015), whereas in humans, maize and wild mice it decays over distances closer to 10,000bp (Chia et al. 2012; Deinum et al. 2015; Genomes Project et al. 2015). It may be, then, that the Sattath test is only applicable when there is relatively short-range LD, such that the patterns of diversity around selected substitutions do not substantially overlap with the analysis windows around neutral ones. If this were the case, interpreting the similarity in troughs of diversity around selected and neutral substitutions as evidence for a paucity of hard selective sweeps may not be justified in organisms where LD decays over distances of a similar order of magnitude as the width of the diversity troughs themselves.
   	        	  
\section{Fitting genome wide patterns}

	Methods to estimate the rate and strength of positive selection in the genome employ various combinations of nucleotide diversity, divergence, recombination rates and estimates of background selection effects as summary statistics, averaged over many regions of the genome. Recently, Elyashiv et al. (2016) developed a method that fits a model of hard sweeps and background selection to genome-wide variation in nucleotide diversity and divergence (at both selected and neutral sites). In D. melanogaster, they showed that hard sweeps can explain a large amount of genome-wide variation genetic diversity. For nonsynonymous sites, they found that α = 4.1\% for strongly selected mutations (s ≥ 0.03\%) and α = 36.3\% for weakly selected mutations (s ~ 0.0003\%), summing to α = 40.4\%, which is similar to the estimate obtained using the MK-test (Andolfatto 2007). Their results suggest that accounting for weakly selected mutations may help reconcile the discrepancy between MK-based estimates of the rate and strength of selection and parameters estimated from sweep model predictions, described above.

Elyashiv et al (2016) showed that a map of the effects of hard sweeps and background selection is capable of explaining a large amount of the variation in diversity across the genome, further demonstrating that the action of natural selection is pervasive, at least in D. melanogaster. However, their method overestimated the rate of deleterious mutations, which the authors attribute to the presence of modes of adaptation other than hard sweeps in D. melanogaster. 
